client/human/HumanClientApp.cpp

```diff
@@ -1316,3 +1316,57 @@ void HumanClientApp::OpenURL(const std::string& url) {
     // execute open command
     system(command.c_str());
 }
+
+void HumanClientApp::BrowsePath(const boost::filesystem::path& browse_path) {
+    if (browse_path.empty() || browse_path == "/") {
+        ErrorLogger() << "Invalid path: " << browse_path.string();
+        return;
+    }
+
+    boost::filesystem::path full_path;
+
+    try {
+        boost::filesystem::file_status status = boost::filesystem::status(browse_path);
+        if (!boost::filesystem::exists(status)) {
+            DebugLogger() << "Non-existant path: " << browse_path.string() << " trying parent directory";
+            if (browse_path.has_parent_path())
+                BrowsePath(browse_path.parent_path());
+            return;
+        }
+
+        // Validate as a canonical path
+        if (boost::filesystem::is_directory(status)) {
+            full_path = boost::filesystem::canonical(browse_path);
+        } else {
+            // If given a file, use the files containing directory
+            DebugLogger() << "Non-directory target: " << browse_path.string() << ", using parent directory";
+            full_path = boost::filesystem::canonical(browse_path.parent_path());
+        }
+
+        // Verify not a regular file
+        if (boost::filesystem::is_regular_file(full_path)) {
+            ErrorLogger() << "Target directory " << full_path.string() << " is a regular file, given path argument: "
+                          << browse_path.string();
+            return;
+        }
+
+    } catch (const boost::filesystem::filesystem_error& ec) {
+        ErrorLogger() << "Filesystem error when attempting to browse directory " << browse_path.string()
+                      << ": " << ec.what();
+        return;
+    }
+
+    if (full_path.empty()) {
+        ErrorLogger() << "Unable to determine directory for path " << browse_path.string();
+        return;
+    }
+
+// Append trailing slash to prevent executing a file with same name(minus extension) as folder
+#if defined(_WIN32)
+    system(("start /b " + PathString(full_path) + "\\").c_str());
+#elif defined(__APPLE__)
+    system(("open " + PathString(full_path) + "/ &").c_str());
```

Comment:
On OSX this does not work when the path contains white spaces. You need to wrap the path in double quotes:
```
    system(("open \"" + PathString(full_path) + "/\" &").c_str());
```
I'd expect the same problem on the other platforms, but didn't test...